src/main/java/org/generationcp/middleware/service/impl/GermplasmGroupingServiceImpl.java

```diff
@@ -43,76 +46,66 @@
 	@Override
-	@Transactional
-	public GermplasmGroup markFixed(final Germplasm germplasmToFix, final boolean includeDescendants, final boolean preserveExistingGroup) {
-		GermplasmGroupingServiceImpl.LOG.info("Marking germplasm with gid {} as fixed.", germplasmToFix.getGid());
-
-		if (includeDescendants) {
-			final GermplasmPedigreeTree tree = this.getDescendantTree(germplasmToFix);
-			this.traverseAssignGroup(tree.getRoot(), germplasmToFix.getGid(), preserveExistingGroup);
-		} else {
-			this.assignGroup(germplasmToFix, germplasmToFix.getGid(), preserveExistingGroup);
+	public List<GermplasmGroup> markFixed(final List<Integer> gids, final boolean includeDescendants, final boolean preserveExistingGroup) {
+		final List<Germplasm> germplasmList = this.daoFactory.getGermplasmDao().getByGIDList(gids);
+		final Map<Integer, Germplasm> germplasmMap = germplasmList.stream().collect(Collectors.toMap(
+			Germplasm::getGid, Function.identity()));
+		for (final Integer gid : gids) {
+			GermplasmGroupingServiceImpl.LOG.info("Marking germplasm with gid {} as fixed.", gid);
+			final Germplasm germplasmToFix = germplasmMap.get(gid);
+			if (includeDescendants) {
+				final GermplasmPedigreeTree tree = this.getDescendantTree(germplasmToFix);
+				this.traverseAssignGroup(tree.getRoot(), gid, preserveExistingGroup);
+			} else {
+				this.assignGroup(germplasmToFix, gid, preserveExistingGroup);
+			}
 		}
-
-		return this.getGroupMembers(germplasmToFix);
+		return this.getGroupMembersForGermplasm(germplasmList);
 	}
 
 	@Override
-	@Transactional
-	public void unfixLines(final Set<Integer> gids) {
-		this.germplasmDAO.resetGermplasmGroup(new ArrayList<Integer>(gids));
+	public List<Integer> unfixLines(final List<Integer> gids) {
+		final List<Integer> gidsToUnFix = new ArrayList<>(gids);
+		final GermplasmDAO germplasmDao = this.daoFactory.getGermplasmDao();
+		final List<Integer> unfixedLines =
+			germplasmDao.getGermplasmWithoutGroup(gids).stream().map(Germplasm::getGid).collect(Collectors.toList());
+		gidsToUnFix.removeAll(unfixedLines);
+		germplasmDao.resetGermplasmGroup(gidsToUnFix);
+		return gidsToUnFix;
 	}
 
 	@Override
-	public GermplasmGroup getGroupMembers(final Germplasm founder) {
-		final GermplasmGroup germplasmGroup = new GermplasmGroup();
+	public List<GermplasmGroup> getGroupMembers(final List<Integer> gids) {
+		final List<Germplasm> germplasmList = this.daoFactory.getGermplasmDao().getByGIDList(gids);
+		return this.getGroupMembersForGermplasm(germplasmList);
+	}
+
+	private List<GermplasmGroup> getGroupMembersForGermplasm(final List<Germplasm> germplasmList) {
+		final List<GermplasmGroup> germplasmGroupList = new ArrayList<>();
+		for (final Germplasm founder : germplasmList) {
+			final GermplasmGroup germplasmGroup = new GermplasmGroup();
 
-		final Method method = this.methodDAO.getById(founder.getMethodId());
-		founder.setMethod(method);
+			final Method method = this.daoFactory.getMethodDAO().getById(founder.getMethodId());
```

Comment:
I think method is already available in germplasm.getMethod(). No need to get it from DAO.